ocaml/xapi/xapi_pool_update.ml

```diff
@@ -302,7 +302,25 @@ let verify update_info update_path =
     ) [update_xml_path; repomd_xml_path];
   debug "Verify signature OK for pool update uuid: %s by key: %s" update_info.uuid update_info.key
 
+let patch_uuid_of_update_uuid uuid =
+  let arr = Uuid.int_array_of_uuid (Uuid.uuid_of_string uuid) in
```

Comment:
I'm still not so happy with discarding 6 of the 16 bytes. Here are a few alternatives we could consider:
* Add 1 (wrapping if needed) to the first 4-bit hex-digit (or the first few).
* Rotation, perhaps one step right, or four left which would turn
 * `a1111111-b222-c333-d444-e55555555555` into
 * `b222c333-d444-e555-5555-5555a1111111`
* Swap the first and last chunks, turning
 * `a1111111-b222-c333-d444-e55555555555` into
 * `e5555555-5555-b222-c333-d444a1111111`

Reversibility is a potentially handy property, but any of the above could be combined with zeroing out a few bytes in the middle if you really want.

Regardless of what transformation we choose, the function should have a comment saying why it is needed, and a brief description (and/or example) of the transformation.